TagsterWebAPI/src/Tagster.Application/Queries/GetProfile/GetProfileWithTagsHandler.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Tagster.Application.Services;
+using Tagster.CQRS.Queries.Handlers;
+using Tagster.Application.Dtos;
+using Tagster.DataAccess.DBContexts;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+
+namespace Tagster.Application.Queries.GetProfile
+{
+    internal sealed class GetProfileWithTagsHandler : IQueryHandler<GetProfileWithTags, ProfileDto>
+    {
+        private readonly ITagsService _tagService;
+        public GetProfileWithTagsHandler(ITagsService tagService)
+        {
+            _tagService = tagService;
+        }
+        private readonly TagsterDbContext _tagsterDb;
+        public GetProfileWithTagsHandler(TagsterDbContext tagsterDb)
+            => _tagsterDb = tagsterDb;
+
+        public async Task<ProfileDto> Handle(GetProfileWithTags request, CancellationToken cancellationToken)
+        {
+            DataAccess.Entities.Profile profileInfo = _tagsterDb
+                .Profiles
+                .Where(profile => profile.Href.Equals(request.Href))
+                .Include(profile => profile.ProfileTags)
+                .Include(profile => profile.Name)
```

Comment:
what is the purpose of using Include on Name and LastName?

Reply:
We want to get all info we own about the user based on our identificator (href), I think that front guys will find a way to use it to their advantage, so I'll just leave it there. If You have more concerns about this code please wait untill next meeting and I'll be happy to discuss it with You ^^